resolwe_bio/processes/import_data/expressions.py

```diff
@@ -0,0 +1,417 @@
+        source = StringField(label="Gene ID source")
+        species = StringField(label="Species")
+        build = StringField(label="Build")
+        feature_type = StringField(label="Feature type")
+
+    def run(self, inputs, outputs):
+        """Run analysis."""
+
+        if not inputs.exp and not inputs.rc:
+            self.error("Please provide raw or/and normalized expression files.")
+
+        elif inputs.exp and not inputs.exp_type:
+            self.error(
+                "Please provide normalization type together with normalized expressions."
+            )
+
+        elif not inputs.exp and inputs.exp_type and inputs.rc:
+            self.error("Please provide raw or/and normalized expression files.")
+
+        elif inputs.rc and not inputs.exp and not inputs.exp_type:
+            rc = inputs.rc.import_file(imported_format="compressed")
+            exp = inputs.rc.import_file(imported_format="compressed")
+            stem = Path(rc).stem
+            supported_extensions = (".txt", ".tab")
+            exp_type = "RAW_COUNT"
+
+        elif inputs.exp and inputs.exp_type and not inputs.rc:
+            exp = inputs.exp.import_file(imported_format="compressed")
+            stem = Path(exp).stem
+            supported_extensions = ".tab"
+            exp_type = inputs.exp_type
+
+        else:
+            rc = inputs.rc.import_file(imported_format="compressed")
+            exp = inputs.exp.import_file(imported_format="compressed")
+            stem = Path(rc).stem
+            supported_extensions = (".txt", ".tab")
+            exp_type = inputs.exp_type
+
+        if not stem.endswith(supported_extensions):
+            self.error(
+                f"The imported file has unsupported file name extension. "
+                f"The supported extensions are {supported_extensions}."
+            )
+        name = stem[:-4]
+
+        # Save the abundance estimates to JSON storage
+        expression_to_storage(infile=exp, outfile="json.txt")
+
+        # Prepare the expression set outputs
+        feature_ids = pd.read_csv(exp, sep="\t", index_col="Gene").index.tolist()
+
+        feature_filters = {
+            "source": inputs.source,
+            "species": inputs.species,
+            "feature_id__in": feature_ids,
+        }
+
+        feature_ids_to_names = {
+            f.feature_id: f.name for f in self.feature.filter(**feature_filters)
+        }
+
+        if inputs.rc and inputs.exp:
+            prepare_expression_set(
+                exp=exp,
+                exp_type=exp_type,
+                feature_dict=feature_ids_to_names,
+                outfile_name=f"{name}_expressions",
+                rc=rc,
+            )
+        else:
+            prepare_expression_set(
+                exp=exp,
+                exp_type=exp_type,
+                feature_dict=feature_ids_to_names,
+                outfile_name=f"{name}_expressions",
+            )
+
+        if inputs.rc:
+            outputs.rc = rc
```

Comment:
In the original process, the allowed file suffix on upload was either .txt or .tab, but the process always saved the files with the suffix .tab (except in the case of expression_set output field, where the suffix is .txt). For consistency, I think the suffix should be changed also in this process, so that if the file to be saved in exp or rc fields is uploaded as .txt, the suffix is converted to .tab before saving the file to the outputs.